decimal_test.go

```diff
@@ -247,133 +253,122 @@ func TestNumberCmp(t *testing.T) {
 }
 
 func TestFromInt(t *testing.T) {
-	assert.Equal(t, Number{5, 0}, FromInt(5))
-	assert.Equal(t, Number{0, 0}, FromInt(0))
-	assert.Equal(t, Number{-5, 0}, FromInt(-5))
+	assert.Equal(t, newDecimal.New(5, 0), newDecimal.NewFromInt(5))
+	assert.Equal(t, newDecimal.New(0, 0), newDecimal.NewFromInt(0))
+	assert.Equal(t, newDecimal.New(-5, 0), newDecimal.NewFromInt(-5))
 }
 
 func TestFromRat(t *testing.T) {
 	tests := []struct {
-		rat *big.Rat
-		exp int
-		n   Number
+		rat      *big.Rat
+		exp      int
+		expected Number
 	}{
 		{
-			rat: big.NewRat(1234, 100),
-			exp: -2,
-			n:   Number{1234, -2},
+			rat:      big.NewRat(1234, 100),
+			exp:      -2,
+			expected: newDecimal.New(1234, -2),
 		},
 		{
-			rat: big.NewRat(1234, 100),
-			exp: -1,
-			n:   Number{123, -1},
+			rat:      big.NewRat(1234, 100),
+			exp:      -1,
+			expected: newDecimal.New(123, -1),
 		},
 		{
-			rat: big.NewRat(1234, 100),
-			exp: 0,
-			n:   Number{12, 0},
+			rat:      big.NewRat(1234, 100),
+			exp:      0,
+			expected: newDecimal.New(12, 0),
 		},
 		{
-			rat: big.NewRat(1234, 100),
-			exp: 1,
-			n:   Number{1, 1},
+			rat:      big.NewRat(1234, 100),
+			exp:      1,
+			expected: newDecimal.New(1, 1),
 		},
 		{
-			rat: big.NewRat(1000000000, 3),
-			exp: 0,
-			n:   Number{333333333, 0},
+			rat:      big.NewRat(1000000000, 3),
+			exp:      0,
+			expected: newDecimal.New(333333333, 0),
 		},
 		{
-			rat: big.NewRat(1000000000, 3),
-			exp: -7,
-			n:   Number{3333333333333333, -7},
+			rat:      big.NewRat(1000000000, 3),
+			exp:      -7,
+			expected: newDecimal.New(3333333333333333, -7),
 		},
 		{
-			rat: big.NewRat(1000000000, 3),
-			exp: -8,
-			n:   Number{33333333333333332, -8},
+			rat:      big.NewRat(1000000000, 3),
+			exp:      -8,
+			expected: newDecimal.New(33333333333333331, -8),
```

Comment:
Seems like the expected output has changed in some cases where higher precision is required. Won't this break anything?

Reply:
Dovydas PR if passed will fix this https://github.com/shopspring/decimal/pull/288/files. 
At the moment `NewFromRat` was ported from https://github.com/shopspring/decimal/pull/4/files which works for small Rats, big.Rats won't necessarily fit into a Float64, so we will be losing information here.